File: src/main/java/me/creepinson/creepinoutils/util/client/gui/CustomList.java
package me.creepinson.creepinoutils.util.client.gui;

import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.widget.list.AbstractList;
import net.minecraft.client.gui.widget.list.ExtendedList;

public class CustomList<E extends AbstractList.AbstractListEntry<E>> extends ExtendedList<E> {

    public CustomList(Minecraft mcIn, int widthIn, int heightIn, int topIn, int bottomIn, int slotHeightIn) {
        super(mcIn, widthIn, heightIn, topIn, bottomIn, slotHeightIn);
    }

    @Override
    protected void renderHoleBackground(int p_renderHoleBackground_1_, int p_renderHoleBackground_2_,
                                        int p_renderHoleBackground_3_, int p_renderHoleBackground_4_) {
        // Do not render dirt background
    }

}

File: src/main/kotlin/com/theoparis/creepinoutils/util/client/gui/CustomList.kt
package com.theoparis.creepinoutils.util.client.gui

import com.mojang.blaze3d.matrix.MatrixStack
import net.minecraft.client.Minecraft
import net.minecraft.client.gui.widget.list.AbstractList
import net.minecraft.client.gui.widget.list.ExtendedList

class CustomList<E : AbstractList.AbstractListEntry<E>>(
    mcIn: Minecraft,
    widthIn: Int,
    heightIn: Int,
    topIn: Int,
    bottomIn: Int,
    slotHeightIn: Int
) : ExtendedList<E>(mcIn, widthIn, heightIn, topIn, bottomIn, slotHeightIn) {
    override fun renderBackground(matrixStack: MatrixStack) {
        // Do not render dirt background
    }
}